### PR TITLE
[examples] Add async task playground example

### PR DESCRIPTION
**Pull request by augustjk**
Repurposed the existing but hidden async task example to use `@lit-labs/task`.

We can add additional examples with future PRs.

direct preview link: https://pr1179-c068b7f---lit-dev-5ftespv5na-uc.a.run.app/playground/#sample=examples/async-task

### CODE REVIEW

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    this._package = (e.target as HTMLInputElement).value;
+  }
+
+  render() {
+    return html`
+      <header>
+        <h1>${this._package}</h1>
+        <a>${npmLogo}</a>
+      </header>
+      <label>
```
Comment (justinfagnani) [inline packages/lit-dev-content/samples/examples/async-task/npm-info.ts:37]: Lets move the input to above the header so all the dynamic output is below it.

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      if (response.status === 200) {
+        return response.json() as Promise<NpmPackage>;
+      } else {
+        throw response.text();
+      }
+    },
+    args: () => [this._package],
+  });
+
+  _onChange(e: Event) {
```
Comment (justinfagnani) [inline packages/lit-dev-content/samples/examples/async-task/npm-info.ts:27]: I'd mvoe this to below the render method

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    #logo {
+      height: 38px;
+      width: auto;
+    }
+  `;
+}
+
+const getPackageUrl = (name: string) => `https://registry.npmjs.org/${name}`;
+
+const npmLogo = html`
```
Comment (justinfagnani) [inline packages/lit-dev-content/samples/examples/async-task/npm-info.ts:82]: Could we find a live URL for this instead?

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    <polygon fill="#fff" points="1,5 3,5 3,2 4,2 4,5 5,5 5,1 1,1 " />
+    <path fill="#fff" d="M6,1v5h2V5h2V1H6z M9,4H8V2h1V4z" />
+    <polygon
+      fill="#fff"
+      points="11,1 11,5 13,5 13,2 14,2 14,5 15,5 15,2 16,2 16,5 17,5 17,1 "
+    />
+  </svg>
+`;
+
+interface NpmPackage {
```
Comment (justinfagnani) [inline packages/lit-dev-content/samples/examples/async-task/npm-info.ts:106]: To simplify this file, I wonder if we want to put npm registry utilities and types into a npm.ts file. fetchWithDelay could do in there too, just named fetchPackage or something.

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  @state()
+  _package = 'lit';
+
+  _fetchNpmInfoTask = new Task(this, {
+    task: async ([pkgName], {signal}) => {
+      if (pkgName === undefined || pkgName === '') {
+        throw new Error('Empty package name');
+      }
+      // Artifical delay for demo purposes
+      await new Promise((r) => setTimeout(r, 1000));
```
[the remark this answers is not in the document]
Comment (augustjk) [reply packages/lit-dev-content/samples/examples/async-task/npm-info.ts]: I went with `fetchPackageInfo` moved to a separate file. Should the name include `WithDelay`? I don't feel the need to specifically call out that it's slow.

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts (hunk 1 above)
Comment (augustjk) [reply packages/lit-dev-content/samples/examples/async-task/npm-info.ts:23]: Nice, no shifting. I like it.

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts (hunk 2 above)
Comment (augustjk) [reply packages/lit-dev-content/samples/examples/async-task/npm-info.ts]: done

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts (hunk 3 above)
Comment (augustjk) [reply packages/lit-dev-content/samples/examples/async-task/npm-info.ts]: done

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts (hunk 4 above)
Comment (augustjk) [reply packages/lit-dev-content/samples/examples/async-task/npm-info.ts]: done

File: packages/lit-dev-content/samples/examples/async-task/npm-info.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
+import {LitElement, css, html} from 'lit';
+import {customElement, state} from 'lit/decorators.js';
+import {Task} from '@lit-labs/task';
```
Comment (justinfagnani) [inline packages/lit-dev-content/samples/examples/async-task/npm-info.ts:3]: We'll need to update this for graduation. cc @rictic

**Review by justinfagnani — APPROVED** [review]
🎉

**Review by AndrewJakubowicz — APPROVED** [review]
Played with the demo in the preview & it's extremely nice! Thank you and great example!

**Pull request merged**